src/gennylib/include/gennylib/PhaseLoop.hpp

```diff
@@ -109,9 +153,20 @@ class IterationCompletionCheck final {
     // referenceStartingPoint time (versus having those in the ActorPhaseIterator). BUT: even the
     // .end() iterator needs an instance of this, so it's weird
 
+    SteadyClock::time_point& _now() {
+        _cachedNow = SteadyClock::now();
+        return _cachedNow;
+    }
```

Comment:
based on what we were looking at earlier today I think the `now()` calls weren't really an important factor here? If so, can we remove this to keep the logic slightly simpler?

Reply:
Done!

I also added a comment in Jira for my latest hypothesis: https://jira.mongodb.org/browse/TIG-1155?focusedCommentId=2131381&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2131381

The actual rate limit is perfectly fine. I'll send an email once this goes in to clarify the expected behavior of the rate limiter.

Reply:
Oh I also checked the naive version of the rate limiter. It is definitely worse than the global rate limiter.

In the IncActor, I was getting 60% of the expected throughput with 1000 threads and 100us sleeps . I believe this is because `sleep_for` only guarantees a lower bound but seems to sleep for a bit longer than the specified duration.